### PR TITLE
feat: fixed text models to be equals to their kernel counterpart

### LINKED ISSUE #805: UIInputText, UIText and TextShape models do not represent the architecture in the SDK
We are forced to do ugly stuff to parse models for `UIInputText` `UIText` and `TextShape` because our architecture in Unity does not represent the one in the SDK.

For example:

`UIInputText` in the SDK:
```
export class UIInputText extends UIShape {
  outlineWidth: number = 0
  outlineColor: Color4 = Color4.Black()
  color: Color4 = Color4.Clear()
  thickness: number = 1
  fontSize: number = 10
  fontWeight: string = 'normal'
  font?: Font
  value: string = ''
  placeholderColor: Color4 = Color4.White()
  placeholder: string = ''
  margin: number = 10
  maxWidth: number = 100
  hTextAlign: string = 'left'
  vTextAlign: string = 'bottom'
  autoStretchWidth: boolean = true
  background: Color4 = Color4.Black()
  focusedBackground: Color4 = Color4.Black()
  textWrapping: boolean = false
  shadowBlur: number = 0
  shadowOffsetX: number = 0
  shadowOffsetY: number = 0
  shadowColor: Color4 = Color4.White()
  paddingTop: number = 0
  paddingRight: number = 0
  paddingBottom: number = 0
  paddingLeft: number = 0
  onTextSubmit: OnTextSubmit | null = null
  onChanged: OnChanged | null = null
  onFocus: OnFocus | null = null
  onBlur: OnBlur | null = null
}
```

vs Unity

```
 public class UIInputText
    {
        new public class Model : UIShape.Model
        {
            public TextShape.Model textModel;
            public string placeholder;
            public Color placeholderColor = Color.white;
            public Color focusedBackground = Color.black;

            public string onTextSubmit;
            public string onChanged;
            public string onFocus;
            public string onBlur;
        }
   }
```

In unity we encapsulate some properties in the `textModel` field that are present in the root of the SDK `UIInputField`.
This forces us to do this ugly double-serialization:
```
public override BaseModel GetDataFromJSON(string json)
{
            Model model = Utils.SafeFromJson<Model>(json);
            model.textModel = Utils.SafeFromJson<TextShape.Model>(json);
            return model;
}
```

Adapt the classes in Unity to be 1 on 1 with the SDK:
- Avoid the double serialization
- Remove `TextShape.Model`
- Adapt the tests

### PR DESCRIPTION
**Pull request by Suduck**
## What does this PR change?

Fixes #805

Before this PR , the TextShape, UIText and UIInputText shared the same text model, but in kernel they have differents attributes for each model. With this PR, now they have the same model as it is implemented in kernel. However, there is some attributes that we are not using in Unity

## How to test the changes?

1. Go to: https://play.decentraland.zone/index.html?renderer=urn:decentraland:off-chain:renderer-artifacts:feat/update-text-model
2. Go to a place where a UIText, UIInputText and TextShape exists  and check that everything work as expected

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by AjimenezDCL — APPROVED** [review]

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Components/TextShape/TextShape.cs [hunk 1]
```diff
@@ -14,6 +14,14 @@ public class TextShape : BaseComponent
         [Serializable]
         public class Model : BaseModel
         {
+
+            //These values exist in the SDK but we are doing nothing with these values
+            public string fontWeight = "normal";
+            public bool resizeToFit = false;
+            public float zIndex = 0;
+            public bool isPickable = false;
```
Comment (menduz) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Components/TextShape/TextShape.cs:22]: Shall we remove them from the SDK?

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Components/UI/UIInputText/UIInputText.cs [hunk 2]
```diff
@@ -12,7 +12,37 @@ public class UIInputText : UIShape<UIInputTextRefContainer, UIInputText.Model>
...
+
+            public Color color = Color.white;
+
+            //These values exist in the SDK but we are doing nothing with these values
+            public float thickness = 1f;
+            public Color placeHolderColor = Color.white;
+            public float maxWidth = 100f;
+            public bool autoStretchWidth = true;
+            public Color background = Color.black;
+            public string fontWeight = "normal";
```
Comment (menduz) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Components/UI/UIInputText/UIInputText.cs:26]: Should we remove them as well?

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Components/TextShape/TextShape.cs (hunk 1 above)
Comment (BrianAmadori) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Components/TextShape/TextShape.cs:22]: Yeah, I think we should remove them even if we want to implement them in the future.

**Review by BrianAmadori — APPROVED** [review]
LGTM

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Components/UI/UIInputText/UIInputText.cs (hunk 2 above)
Comment (Suduck) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Components/UI/UIInputText/UIInputText.cs:26]: Yes, there is unused properties in UIText, UIInputText and TextShape. We should remove them in the SDK or implement the functionality in Unity

**Pull request merged**